src/promptflow/promptflow/contracts/tool.py

```diff
@@ -123,6 +124,19 @@ def is_connection_value(val):
         val = type(val) if not isinstance(val, type) else val
         return val in connections.values()
 
+    @staticmethod
+    def is_builtin_connection(type_name: str):
+        import promptflow.connections
+
+        for name, obj in inspect.getmembers(promptflow.connections):
+            if inspect.isclass(obj) and name == type_name:
+                return True
+        return False
+
+    @staticmethod
+    def is_custom_strong_type_connection(val):
+        return ConnectionType.is_connection_value(val) and not ConnectionType.is_builtin_connection(val.__name__)
```

Comment:
vector_db connections also not included in promptflow.connections (which mean not builtin connection), the condition may not enough for custom strong type connection

Reply:
Updated to use the base class "CustomStrongTypeConnection" to check